### ISSUE TITLE
PAGER, PAGER_FLAGS override: sometimes hangs

### ISSUE THREAD

**Issue by cbm755** [catch22/octave-doctest#41]
I'm not exactly how reproducible this is...

If I enable the PAGER etc hack that is commented out, and generate some sort of error condition during the doctests, then I never get my prompt back.  If I ctrl-C, I sometimes get:

````
KeyboardInterrupt

warning: connection to external pager lost (pid = 12783)
warning: flushing pending output (please wait)
````
Other times I get just get `KeyboardInterrupt` and my prompt back.

**How to reproduce:**
At least on my system:
  1. start a new Octave instance, load the `symbolic` pkg
  2. `syms x`    (Symbolic spews some stdout and stderr at first start while connecting to Python.  *Leave this step out to reproduce*.)
  3. `doctest symfun`.

*Without* step 2, the stdout/stderr noise from symbolic startup makes the first doctest fail (this is understandable) but it also activates this bug:

````
octave:2> doctest('bernoulli', 'sym')
Doctest v0.3.0-dev: this is Free Software without warranty, see source.
==========================================================================
Start of temporary output (github.com/catch22/octave-doctest/issues/6)
==========================================================================
Python 2.7.9 (default, Apr 15 2015, 12:08:00) 
[GCC 5.0.0 20150319 (Red Hat 5.0.0-0.21)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> 



^C
KeyboardInterrupt
>>> 
octave:2> 
````

**Comment by cbm755** [comment]
Hmm, maybe similar things can happen with Octave `runtests` (I just ctrl-C'd and didn't get my prompt back).  Maybe this is not a big deal.

**Comment by cbm755** [comment]
I don't know if this is only related to symbolic package or not.  symbolic is funny because it has a popen pipe open to python and is getting back both stdout and stderr from that...

**Comment by oheim** [comment]
I cannot reproduce easily. Since standard Octave error handling seems to be okay: What do you mean by “error condition”? Segfaults?

**Comment by cbm755** [comment]
> you mean by “error condition”

I meant having a doctest fail (not match).  But I'd have to experiment some more to try to be precise as I only have the one recipe for reproducing.

symbolic is kind of sensitive to ctrl-c anyway: occasionally the ctrl-c can get through to python.

**Comment by cbm755** [comment]
I wondered if that had something to do with symbolic doing `fflush(stdout)` during startup, but that seems ok in a standalone test.

**Comment by cbm755** [comment]
This is not as serious as I thought.

1.  It is true that I never get my prompt back using the recipe above, assuming I just wait...

2.  However, ctrl-c *does work*.  I've tried it about 30 times and only once did I get `warning: connection to external pager lost (pid = 12783)` and even in that case I did get my prompt back.

So at least symbolic should not hold back kludges contemplated in #6 ...

**Comment by catch22** [comment]
Hmm. From my point of view we can give the pager hack another try, but we don't have to.

**Comment by cbm755** [comment]
> Hmm. From my point of view we can give the pager hack another try, but we don't have to.

Ok with me, do it in a PR and we'll see if travis output looks ok?

**Comment by catch22** [comment]
OK.

**Comment by catch22** [comment]
Closing as this is no longer relevant given @oheim's excellent `evalc` implementation.
